### PR TITLE
coqPackages: etc

### PR DESCRIPTION
**Pull request by CohenCyril**
###### Motivation for this change

Fixes in coqPackages:
- updated coqPackage.heq (broken url)
- fixed use of `DESTDIR` and `COQMF_COQLIB` in mkCoqDerivation

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by CohenCyril** [comment]
@GrahamcOfBorg build coqPackages coqPackages_8_12 coqPackages_8_13

**Comment by CohenCyril** [comment]
@GrahamcOfBorg build coqPackages_8_6 coqPackages_8_7 coqPackages_8_8 coqPackages_8_9 coqPackages_8_10

**Comment by CohenCyril** [comment]
@GrahamcOfBorg build coqPackages_8_7 coqPackages_8_8 coqPackages_8_9 coqPackages_8_10

File: pkgs/build-support/coq/default.nix [hunk 1]
```diff
@@ -77,9 +84,8 @@ stdenv.mkDerivation (removeAttrs ({
 (optionalAttrs setCOQBIN { COQBIN = "${coq}/bin/"; }) //
 (optionalAttrs (!args?installPhase && !args?useMelquiondRemake) {
   installFlags =
-    [ "${var-coqlib-install}=$(out)/lib/coq/${coq.coq-version}/" ] ++
-    optional (match ".*doc$" (args.installTargets or "") != null)
-      "DOCDIR=$(out)/share/coq/${coq.coq-version}/" ++
+    [ "DESTDIR=$(out)" ] ++ [ coqlib-flag ]++
```
Comment (SuperSandro2000) [inline pkgs/build-support/coq/default.nix:87]:
```suggestion
    [ "DESTDIR=$(out)" ] ++ [ coqlib-flag ] ++
```

**Comment by CohenCyril** [comment]
@GrahamcOfBorg build coqPackages_8_7 coqPackages_8_8 coqPackages_8_9 coqPackages_8_10

**Comment by CohenCyril** [comment]
@GrahamcOfBorg build coqPackages_8_5 coqPackages_8_6

**Comment by CohenCyril** [comment]
@GrahamcOfBorg build coqPackages_8_11 coqPackages_8_12 coqPackages_8_13

**Pull request closed without merging**